✨ Introduce getConditionFieldsForProjects

s0 opened the pull request:
This supports HPC-6264

(this will cause type-error conflict with #66, so whichever is merged last will need to be rebased and fixed)

Pl217 commented on src/lib/data/projects.ts:
```diff
@@ -167,3 +169,38 @@ export async function getGoverningEntityIDsForProjects({
+  projects,
+}: {
+  database: Database;
+  projects: Map<ProjectId, ProjectData>;
```
What do you think about making this method easier to use outside your use-case, by requiring less project data?
```suggestion
  projects: Map<ProjectId, Pick<ProjectData, 'projectVersionPlan'>>;
```

Pl217 reviewed (commented):
It makes sense to make these changes in two separate commits, because changes are pretty different. In first commit, extend `getAllProjectsForPlan` and in second commit introduce `getConditionFieldsForProjects`

s0 committed:
✨ Extend `getAllProjectsForPlan` to also return `projectVersionPlan`

s0 committed:
✨ Introduce getConditionFieldsForProjects

s0 committed:
✨ Allow for more flexible use of project lib module

To allow for code that has retreived projects in a different manner to
use the utility functions for loading project data, only require the
needed data to be supplied.

s0 commented:
Rebased ontop of `develop` now #66 is merged, this is ready for review now.

No need to bump the version with this one, as there is a bit more work that is still needed for UN-OCHA/hpc_service#2514
